charts/common/templates/_resources.tpl

```diff
@@ -0,0 +1,22 @@
+{{/*
+Return a resource request/limit object based on a given preset or provided resources.
+{{ include "common.resources" (dict "resources" (dict) "resourcesPreset" "small") -}}
+*/}}
+{{- define "common.resources" -}}
+  {{- $resources := dict -}}
+  {{- if .resources -}}
+    {{- $resources = .resources | merge $resources -}}
+  {{- end -}}
+  {{- if and .resourcesPreset (ne .resourcesPreset "none") -}}
+    {{- $resources = include "common.resources.preset" (dict "type" .resourcesPreset) | fromYaml | merge $resources -}}
```

Comment:
I don't get where the resources are coming from.  
So, i've seen the sizes in the file below, but how does the resource object actually look like?

Reply:
Or is this something the bitnami charts just understand?

Reply:
Ok, nvm, found it (my google fu wasn't strong enough). This is genius: https://github.com/bitnami/charts/blob/main/bitnami/common/templates/_resources.tpl#L16